src/templates/defaultTemplate.js

```diff
@@ -0,0 +1,59 @@
+// @flow
+
+import type {
+  ChangelogType,
+  VersionChangelogType,
+  TemplateHelpersType,
+  EntryType,
+  EntryKindType
+} from '../types';
+
+const entryKindToReadable = {
+  change: 'Changed',
+  addition: 'Added',
+  fix: 'Fixed'
+};
+
+export default function defaultTemplate(
+  helpers: TemplateHelpersType,
+  changelog: ChangelogType
+) {
+  return [
+    '# Changelog',
+    ...changelog.map((versionChangelog) => renderVersionChangelog(helpers, versionChangelog))
+  ].join('\n\n');
+}
+
+function renderVersionChangelog(
+  helpers: TemplateHelpersType,
+  { version, entries }: VersionChangelogType
+): string {
+  // $FlowFixMe: Object.keys() does not refine correctly based on type definition of keys
+  const entryKeys: EntryKindType[] = Object.keys(entries);
+
+  return [
+    `## Version \`${helpers.stringifyVersion(version)}\``,
+    ...entryKeys.map((entryKind) => renderEntriesOfKind(helpers, entryKind, entries[entryKind]))
+  ].join('\n\n');
+}
+
+function renderEntriesOfKind(
+  helpers: TemplateHelpersType,
+  kind: EntryKindType,
+  entries: EntryType[]
+): string {
+  if (entries.length === 0) {
+    return '';
+  }
+
+  return [
+    `### ${entryKindToReadable[kind]}`,
+    ...entries.map(({ component, description }) => {
```

Comment:
Hmm, so my personal feeling: The handlebars template was "explicit", while these render methods are "clever". If you removed handlebars because they had to deal with a couple of null checks, I'm wondering: why? :-)